service/pager/query_pagers.cc

```diff
@@ -401,7 +401,7 @@ void query_pager::handle_result(
         _max = _max - row_count;
         _exhausted = (row_count < page_size && !results->is_short_read()) || _max == 0;
 
-        if (!_exhausted && row_count > 0) {
+        if (!_exhausted) {
```

Comment:
What happens in a mixed cluster when talking to old replicas? I guess nothing bad (removed check it redundant), but let's think about it (and document in changelog).

Reply:
There are not empty pages here, in an old or mixed cluster. To be honest I don't really know why this check existed here at all. Probably as a sanity check.